ctru-rs/src/applets/mii_selector.rs

```diff
@@ -0,0 +1,190 @@
+        // unlikely and is documented
+        let c_text = CString::new(text).expect("Failed to convert the title text into a CString");
+        unsafe {
+            ctru_sys::miiSelectorSetTitle(self.config.as_mut(), c_text.as_ptr());
+        }
+    }
+
+    /// Set the options of the Mii Selector
+    pub fn set_options(&mut self, options: Options) {
+        unsafe { ctru_sys::miiSelectorSetOptions(self.config.as_mut(), options.bits) }
+    }
+
+    /// Whitelist a guest Mii
+    pub fn whitelist_guest_mii(&mut self, mii_index: MiiConfigIndex) {
+        let index = match mii_index {
+            MiiConfigIndex::Index(i) => i,
+            MiiConfigIndex::All => ctru_sys::MIISELECTOR_GUESTMII_SLOTS,
+        };
+
+        unsafe { ctru_sys::miiSelectorWhitelistGuestMii(self.config.as_mut(), index) }
+    }
+
+    /// Blacklist a guest Mii
+    pub fn blacklist_guest_mii(&mut self, mii_index: MiiConfigIndex) {
+        let index = match mii_index {
+            MiiConfigIndex::Index(i) => i,
+            MiiConfigIndex::All => ctru_sys::MIISELECTOR_GUESTMII_SLOTS,
+        };
+
+        unsafe { ctru_sys::miiSelectorBlacklistGuestMii(self.config.as_mut(), index) }
+    }
+
+    /// Whitelist a user Mii
+    pub fn whitelist_user_mii(&mut self, mii_index: MiiConfigIndex) {
+        let index = match mii_index {
+            MiiConfigIndex::Index(i) => i,
+            MiiConfigIndex::All => ctru_sys::MIISELECTOR_USERMII_SLOTS,
+        };
+
+        unsafe { ctru_sys::miiSelectorWhitelistUserMii(self.config.as_mut(), index) }
+    }
+
+    /// Blacklist a user Mii
+    pub fn blacklist_user_mii(&mut self, mii_index: MiiConfigIndex) {
+        let index = match mii_index {
+            MiiConfigIndex::Index(i) => i,
+            MiiConfigIndex::All => ctru_sys::MIISELECTOR_USERMII_SLOTS,
+        };
+
+        unsafe { ctru_sys::miiSelectorBlacklistUserMii(self.config.as_mut(), index) }
+    }
+
+    /// Set where the cursor will be.
+    /// If there's no Mii at that index, the cursor will start at the Mii with the index 0
+    pub fn set_initial_index(&mut self, index: u32) {
+        // This function is static inline in libctru
+        // https://github.com/devkitPro/libctru/blob/af5321c78ee5c72a55b526fd2ed0d95ca1c05af9/libctru/include/3ds/applets/miiselector.h#L155
+        self.config.initial_index = index
+    }
+
+    /// Launch the Mii Selector.
+    /// Returns an error when the checksum of the Mii is invalid.
+    pub fn launch(&mut self) -> Result<MiiSelectorReturn, MiiLaunchError> {
+        let mut return_val = Box::<ctru_sys::MiiSelectorReturn>::default();
+        unsafe { ctru_sys::miiSelectorLaunch(self.config.as_mut(), return_val.as_mut()) }
+
+        if unsafe { ctru_sys::miiSelectorChecksumIsValid(return_val.as_mut()) } {
+            Ok(return_val.into())
+        } else {
+            Err(MiiLaunchError::InvalidChecksum)
+        }
+    }
+}
+
+impl From<Box<ctru_sys::MiiSelectorReturn>> for MiiSelectorReturn {
```

Comment:
nit: this should be `From<ctru_sys::MiiSelectorReturn>`. The `Box` part isn't needed/utilized.